### PR TITLE
Make the `featureLevel` not optional in the user's profile (OSIO#1937)

### PR DESCRIPTION
**Pull request by xcoulon**
Make the field in the `user` type as a `string` instead of `*string`.
Make the DB column `NOT NULL` in the DB with a default value to `released`,
and update all existing records.

Also, refactored some code to reduce the amount of lines to init user/indentity.

Fixes openshiftio/openshift.io#1937

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: account/user.go [hunk 1]
```diff
@@ -31,13 +31,18 @@ type User struct {
 	Bio                string             // The bio of the User
 	URL                string             // The URL of the User
 	Company            string             // The (optional) Company of the User
-	FeatureLevel       *string            // the (optional) level of features that the user opted in (to access unreleased features). Defaults to `released` so no non-released feature is enabled for the user.
+	FeatureLevel       string             // the (optional) level of features that the user opted in (to access unreleased features). Defaults to `released` so no non-released feature is enabled for the user.
```
Comment (alexeykazakov) [inline account/user.go]: Fix the comment pleas. It's not optional anymore :)

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-279-1`

File: controller/users_blackbox_test.go [hunk 2]
```diff
@@ -1206,20 +1037,23 @@ func WithEmailAddressVerified(verified bool) CreateUserOption {
...
 		Cluster:      "My OSO cluster url",
-		EmailPrivate: false, // being explicit
+		EmailPrivate: false,                       // being explicit
+		FeatureLevel: account.DefaultFeatureLevel, // being explicit
 	}
 	for _, option := range options {
 		option(&user)
 	}
-	return user
+	identity, err := testsupport.CreateTestUser(s.DB, &user)
```
Comment (sbose78) [inline controller/users_blackbox_test.go]: nice, thanks!

File: account/user.go (hunk 1 above)
Comment (xcoulon) [reply account/user.go]: oh yes, you're right! thanks for spotting that.

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-279-2`

**Comment by xcoulon** [comment]
thanks for the review and approval, @sbose78 and @alexeykazakov!

**Pull request merged**